src/analyzer/analyzerbeats.cpp

```diff
@@ -237,10 +237,12 @@ void AnalyzerBeats::storeResults(TrackPointer tio) {
     } else {
         float bpm = m_pPlugin->getBpm();
         qDebug() << "AnalyzerBeats plugin detected constant BPM: " << bpm;
-        pBeats = BeatFactory::makeBeatGrid(*tio, bpm, 0.0f);
+        pBeats = mixxx::BeatsPointer(new mixxx::Beats(tio.get()));
```

Comment:
`std::make_shared<mixxx::BeatsPointer>(...)`? Please avoid `operator new` if possible.

Reply:
I can do it.
Just to mention that I will like to understand why.
We have two option to code that:
1.`pBeats = std::make_shared<mixxx::Beats>(mixxx::Beats(pTrack.get()));`
2.`pBeats = mixxx::BeatsPointer(new mixxx::Beats(pTrack.get()));`

AFAIK, from the pure technical point of view 1 has a minor efficiency advantage due to the fact that only one heap allocation happens while in 2 there are two allocations. There used to be an issue related to exception safety but this has been solved in C++17. There is a nice explanation in [StackOverflow](https://stackoverflow.com/questions/20895648/difference-in-make-shared-and-normal-shared-ptr-in-c). So basically I see no risk on version 2, is not really a naked pointer as I'm giving it immediately to be processed/protected.

But from the point of view of readability of code, 2 is giving me a clear indication that I'm creating a BeatsPointer from a Beats, while in 1 I must know that BeatsPointer is a std::shared_pointer<Beats>. So 2 is my preferred version.

That said, I have no problem on going one way or the other. I will try to be consistent and use always the same. Can I have a comment on that?

Thanks.

Reply:
done